compiler/noirc_frontend/src/elaborator/comptime.rs

```diff
@@ -89,6 +89,39 @@ impl<'context> Elaborator<'context> {
         result
     }
 
+    pub fn elaborate_item_in_module<'a, T>(
+        &'a mut self,
+        module: ModuleId,
+        f: impl FnOnce(&mut Elaborator<'a>) -> T,
+    ) -> T {
+        // Create a fresh elaborator to ensure no state is changed from
+        // this elaborator
+        let mut elaborator = Elaborator::new(
```

Comment:
Can you refactor this so that this and `elaborate_item_from_comptime` share a common helper method? Probably with an optional function and module? Although I guess it doesn't make sense to have both. Maybe a lambda for setting up the fresh elaborator then?